src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs

```diff
@@ -74,28 +82,24 @@ public T DeserializePayload<T>(Message message)
         {
             T retValue = default(T);
 
-            // TODO: Currently we use json serializer auto only for non-testmessage types
-            // CHECK: Can't we just use auto for everything
-            if (MessageType.TestMessage.Equals(message.MessageType))
-            {
-                retValue = message.Payload.ToObject<T>();
-            }
-            else
-            {
-                retValue = message.Payload.ToObject<T>(serializer);
-            }
+            var versionedMessage = message as VersionedMessage;
+            var serializer = versionedMessage?.Version == 2 ? payloadSerializer2 : payloadSerializer;
```

Comment:
can we have a `GetPayloadSerializer(Constants.CurrentVersion)` for this? This can then be reused everywhere here.